### PR TITLE
Fix minimum penalty charge

### PR DESCRIPTION
**Pull request by mpapierski**
Co-authored-by: Ed Hastings <ed@casperlabs.io>
Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>
Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>

### CODE REVIEW

File: smart_contracts/contracts/test/do-nothing-stored-upgrader/src/main.rs [hunk 1]
```diff
@@ -24,6 +24,7 @@ const ARG_PURSE_NAME: &str = "purse_name";
 
 #[no_mangle]
 pub extern "C" fn delegate() {
+    let _named_keys = runtime::list_named_keys();
```
Comment (Fraser999) [inline smart_contracts/contracts/test/do-nothing-stored-upgrader/src/main.rs:27]: Should comment why this is necessary while the contract is still named "do-nothing".  Same for other two similar contracts.

File: execution_engine_testing/tests/src/test/regression/regression_20210924.rs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        proposer_balance_after
+    );
+}
+
+#[ignore]
+#[test]
+fn should_execute_do_minimum_session() {
+    let minimum_deploy_payment = U512::from(DEFAULT_NOP_COST);
+
+    let do_nothing_request = {
```
Comment (Fraser999) [inline execution_engine_testing/tests/src/test/regression/regression_20210924.rs]:
```suggestion
    let do_minimum_request = {
```

File: execution_engine/src/core/engine_state/mod.rs [hunk 3]
```diff
@@ -1576,6 +1602,29 @@ where
             );
         }
 
+        // Session execution was zero cost.
+        // Check if the payment purse can cover the minimum floor for session execution.
+        if session_result.cost().is_zero() && payment_purse_balance < max_payment_cost {
+            // When session code structure is invalid but still has 0 cost we should propagate the
```
Comment (Fraser999) [inline execution_engine/src/core/engine_state/mod.rs]:
```suggestion
            // When session code structure is valid but still has 0 cost we should propagate the
```

File: smart_contracts/contracts/test/do-nothing-stored-upgrader/src/main.rs (hunk 1 above)
Comment (EdHastingsCasperAssociation) [reply smart_contracts/contracts/test/do-nothing-stored-upgrader/src/main.rs:27]: should rename the do-nothing family of contracts to "do-nothing-useful..." or "exercise-lower-gas-bounds..." or something like that

**Review by EdHastingsCasperAssociation — APPROVED** [review]
optimistic approval, once nitpicks have been considered

**Review by Fraser999 — APPROVED** [review]
Optimistic approval given the nitpicks are addressed.

File: smart_contracts/contracts/test/do-nothing-stored-upgrader/src/main.rs (hunk 1 above)
Comment (mpapierski) [reply smart_contracts/contracts/test/do-nothing-stored-upgrader/src/main.rs:27]: I agree - I'll address it adequately in a separate ticket

**Pull request merged**